gnovm/pkg/gnomod/pkg.go

```diff
@@ -165,3 +167,94 @@ func (sp SortedPkgList) GetNonDraftPkgs() SortedPkgList {
 	}
 	return res
 }
+
+// SubPkgsFromPaths returns a list of subpackages from the given paths.
+func SubPkgsFromPaths(paths []string) ([]*SubPkg, error) {
+	for _, path := range paths {
+		fi, err := os.Stat(path)
+		if err != nil {
+			return nil, err
+		}
+		if fi.IsDir() {
+			continue
+		}
+		if filepath.Ext(path) != ".gno" {
+			return nil, fmt.Errorf("files must be .gno files: %s", path)
+		}
+
+		subPkg, err := GnoFileSubPkg(paths)
+		if err != nil {
+			return nil, err
+		}
+		return []*SubPkg{subPkg}, nil
+	}
+
+	subPkgs := make([]*SubPkg, 0, len(paths))
+	for _, path := range paths {
+		subPkg := newEmptySubPkg()
+
+		matches, err := filepath.Glob(filepath.Join(path, "*.gno"))
+		if err != nil {
+			return nil, fmt.Errorf("failed to match pattern: %w", err)
+		}
+
+		subPkg.Dir = path
+		for _, match := range matches {
+			if strings.HasSuffix(match, "_test.gno") {
+				subPkg.TestGnoFiles = append(subPkg.TestGnoFiles, match)
+				continue
+			}
+
+			if strings.HasSuffix(match, "_filetest.gno") {
+				subPkg.FiletestGnoFiles = append(subPkg.FiletestGnoFiles, match)
+				continue
+			}
+			subPkg.GnoFiles = append(subPkg.GnoFiles, match)
+		}
+
+		subPkgs = append(subPkgs, subPkg)
+	}
+
+	return subPkgs, nil
+}
+
+// GnoFileSubPkg returns a subpackage from the given .gno files.
+func GnoFileSubPkg(files []string) (*SubPkg, error) {
```

Comment:
This method might be private, it is only used on `SubPkgsFromPaths`

Reply:
I plan to use this function in later PRs. 

> You can consider this as a preparatory PR in the `gno test` improvement series :)